opennms-dao-api/src/main/java/org/opennms/netmgt/dao/util/ReductionKeyHelper.java

```diff
@@ -34,13 +34,27 @@
 
 import org.opennms.core.utils.InetAddressUtils;
 import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.model.OnmsApplication;
 import org.opennms.netmgt.model.OnmsMonitoredService;
 
 public class ReductionKeyHelper {
 
     // TODO: The distributed poller name (now monitoring system name?) should be part of the edge details
     public static final String DEFAULT_DISTRIBUTED_POLLER_NAME = "";
 
+    public static Set<String> getReductionKeys(final OnmsApplication application) {
+        Objects.requireNonNull(application);
+        Set<String> reductionKeys = new HashSet<>();
+
+        for(OnmsMonitoredService monitoredService : application.getMonitoredServices()) {
```

Comment:
Can we not duplicate the next 3 lines of code? Use the `getReductionKeys(OnmsMonitoredService)` method instead.

Reply:
Yes, right